### ISSUE TITLE
make inventory_dir be available to vars_files resolution/templating

### ISSUE THREAD

**Issue by willthames** [ansible/ansible#9723]
##### Issue Type:

Bug Report
##### Ansible Version:

```
ansible 1.9 (devel 3b80f63e22) last updated 2014/12/05 14:09:09 (GMT +1000)
  lib/ansible/modules/core: (ec2_snapshot_remove b287e7c0a6) last updated 2014/12/02 11:18:53 (GMT +1000)
  lib/ansible/modules/extras: (detached HEAD 7e6fc7023d) last updated 2014/11/05 11:09:14 (GMT +1000)
  v2/ansible/modules/core: (detached HEAD cb69744bce) last updated 2014/11/05 11:09:23 (GMT +1000)
  v2/ansible/modules/extras: (detached HEAD 8a4f07eecd) last updated 2014/11/05 11:09:31 (GMT +1000)
  configured module search path = None
```
##### Environment:

Fedora 21
##### Summary:

You cannot use `inventory_dir` with `vars_files` - it's just not available to the `_update_vars_files` method. 
##### Steps To Reproduce:

```

---
- hosts: localhost

  vars_files:
    - "{{included}}"

  tasks:
    - debug: msg="{{included}}"
    - debug: msg="{{inventory_dir}}"
    - debug: msg="{{hello}}"
```

run with `ansible-playbook -i ~/hosts` and `~/files/hello` containing:

```
hello: world
```

and `~/group_vars/all.yml` containing `included: "{{inventory_dir}}/files/hello`
shows that `inventory_dir` is correctly set but `hello` which should be set by `files/hello` is never set. 
##### Expected Results:

`vars_files` should be able to expand `inventory_dir`
##### Actual Results:

```
PLAY [localhost] ************************************************************** 

GATHERING FACTS *************************************************************** 
ok: [localhost]

TASK: [debug msg="{{included}}"] ********************************************** 
ok: [localhost] => {
    "msg": "/home/will/files/hello"
}

TASK: [debug msg="/home/will"] ********************************************* 
ok: [localhost] => {
    "msg": "/home/will"
}

TASK: [debug msg="{{hello}}"] ************************************************* 
fatal: [localhost] => One or more undefined variables: 'hello' is undefined

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/home/will/vars_files.retry

```

**Comment by hkariti** [comment]
This is still a bug in 1.8.4 and latest dev. It does behave correctly with imported playbooks however.

**Comment by jgornick** [comment]
Duplicate?: #10009

**Comment by jgornick** [comment]
Comment from #10009:

I'd like to add that `playbook_dir` is also unavailable. I can confirm this used to work in 1.7.2.

```
ansible-playbook -i env/prod/hosts -c local playbook.yml 
```

`env/prod/hosts`

```
localhost
```

`env/prod/vars.yml`

```
---
inventory_vars:
  foo: bar
```

`vars.yml`

```
---
playbook_vars:
  foo: bar
```

`playbook.yml`

```
- name: Playbook and Inventory Variables
  hosts: all

  vars_files:
    # Doesn't evaluate playbook_dir and inventory_dir
    - "{{ playbook_dir }}/vars.yml"
    - "{{ inventory_dir }}/vars.yml"
    # Works
    # - vars.yml
    # - env/prod/vars.yml

  tasks:
    - debug:
        var: playbook_dir
    - debug:
        var: inventory_dir
    - debug:
        var: playbook_vars
    - debug:
        var: inventory_vars
    - assert:
        that:
          - "playbook_vars is defined"
          - "inventory_vars is defined"
```

**Comment by bcoca** [comment]
no need to copy comments as the other tickets are linked and will not be deleted

**Comment by hkariti** [comment]
still happens on latest devel

also, how is this a feature idea?

**Comment by moises-silva** [comment]
@hkariti I've made a PR. If you can give my branch a go and report results that'd be nice. The fix has worked for me, but I'm new to ansible so not sure if it breaks something else (the fix seems safe though, and passes all the automated tests).

**Comment by hkariti** [comment]
@moises-silva Thanks! I'll report back

**Comment by jimi-c** [comment]
This is resolved in the current devel and stable-2.1, so we will go ahead and close this now.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
